packages/editor-ui/src/composables/useCanvasOperations.ts

```diff
@@ -0,0 +1,215 @@
+
+			if (trackBulk) {
+				historyStore.stopRecordingUndo();
+			}
+		}
+	}
+
+	function deleteNode(id: string, { trackHistory = false, trackBulk = true } = {}) {
+		const node = workflowsStore.getNodeById(id);
+		if (!node) {
+			return;
+		}
+
+		if (trackHistory && trackBulk) {
+			historyStore.startRecordingUndo();
+		}
+
+		workflowsStore.removeNodeById(id);
+		workflowsStore.removeNodeConnectionsById(id);
+		workflowsStore.removeNodeExecutionDataById(id);
+
+		if (trackHistory) {
+			historyStore.pushCommandToUndo(new RemoveNodeCommand(node));
+
+			if (trackBulk) {
+				historyStore.stopRecordingUndo();
+			}
+		}
+
+		trackDeleteNode(id);
+	}
+
+	function revertDeleteNode(node: INodeUi) {
+		workflowsStore.addNode(node);
+	}
+
+	function trackDeleteNode(id: string) {
+		const node = workflowsStore.getNodeById(id);
+		if (!node) {
+			return;
+		}
+
+		if (node.type === STICKY_NODE_TYPE) {
+			telemetry.track('User deleted workflow note', {
+				workflow_id: workflowsStore.workflowId,
+				is_welcome_note: node.name === QUICKSTART_NOTE_NAME,
+			});
+		} else {
+			void externalHooks.run('node.deleteNode', { node });
+			telemetry.track('User deleted node', {
+				node_type: node.type,
+				workflow_id: workflowsStore.workflowId,
+			});
+		}
+	}
+
+	/**
+	 * Connection operations
+	 */
+
+	function createConnection(connection: Connection) {
+		const sourceNode = workflowsStore.getNodeById(connection.source);
+		const targetNode = workflowsStore.getNodeById(connection.target);
+		if (!sourceNode || !targetNode || !checkIfNodeConnectionIsAllowed(sourceNode, targetNode)) {
+			return;
+		}
+
+		const mappedConnection = mapCanvasConnectionToLegacyConnection(
+			sourceNode,
+			targetNode,
+			connection,
+		);
+		workflowsStore.addConnection({
+			connection: mappedConnection,
+		});
+
+		uiStore.stateIsDirty = true;
+	}
+
+	function deleteConnection(
+		connection: Connection,
+		{ trackHistory = false, trackBulk = true } = {},
+	) {
+		const sourceNode = workflowsStore.getNodeById(connection.source);
+		const targetNode = workflowsStore.getNodeById(connection.target);
+		if (!sourceNode || !targetNode) {
+			return;
+		}
+
+		const mappedConnection = mapCanvasConnectionToLegacyConnection(
+			sourceNode,
+			targetNode,
+			connection,
+		);
+
+		if (trackHistory && trackBulk) {
+			historyStore.startRecordingUndo();
+		}
+
+		workflowsStore.removeConnection({
+			connection: mappedConnection,
+		});
+
+		if (trackHistory) {
+			historyStore.pushCommandToUndo(new RemoveConnectionCommand(mappedConnection));
+
+			if (trackBulk) {
+				historyStore.stopRecordingUndo();
+			}
+		}
+	}
+
+	function revertDeleteConnection(connection: [IConnection, IConnection]) {
+		workflowsStore.addConnection({
+			connection,
+		});
+	}
+
+	// @TODO Figure out a way to improve this
+	function checkIfNodeConnectionIsAllowed(_sourceNode: INodeUi, _targetNode: INodeUi): boolean {
```

Comment:
```suggestion
	function isConnectionAllowed(_sourceNode: INodeUi, _targetNode: INodeUi): boolean {
```

I know this has a TODO but at-least checking if `sourceNode.id !== targetNode.id;` could be a good start :D

Reply:
Good suggestion. Updated.